test/testutils/util.go

```diff
@@ -306,20 +307,41 @@ func CheckLeaderWorkerSetHasCondition(ctx context.Context, k8sClient client.Clie
 	return false, nil
 }
 
-func HasTPUEnvVarsPopulated(pod corev1.Pod) bool {
+func hasAnyEnvVarPopulated(pod corev1.Pod, envVars []string) bool {
 	var containers []corev1.Container
 	containers = append(containers, pod.Spec.Containers...)
 	containers = append(containers, pod.Spec.InitContainers...)
 	for _, container := range containers {
 		for _, env := range container.Env {
-			if env.Name == acceleratorutils.TpuWorkerHostNames || env.Name == acceleratorutils.TpuWorkerId {
-				return true
+			for _, e := range envVars {
+				if env.Name == e {
+					return true
+				}
```

Comment:
the logic here should be "the pod having all the required env vars" right? but right now it only checks one env var of the envVars list exists?